Extend/Controller/Admin/PaymentMain.php

```diff
@@ -0,0 +1,112 @@
+ * https://github.com/wirecard/oxid-ee/blob/master/LICENSE
+ */
+
+namespace Wirecard\Oxid\Extend\Controller\Admin;
+
+use OxidEsales\Eshop\Core\Registry;
+use Wirecard\Oxid\Extend\Model\Payment;
+use Wirecard\Oxid\Model\SofortPaymentMethod;
+use Wirecard\PaymentSdk\Config\Config;
+use Wirecard\PaymentSdk\TransactionService;
+
+/**
+ * Controls the payment method config view.
+ *
+ * @mixin \OxidEsales\Eshop\Application\Controller\Admin\PaymentMain
+ */
+class PaymentMain extends PaymentMain_parent
+{
+    /**
+     * @inheritdoc
+     *
+     * @return string
+     */
+    public function render()
+    {
+        $sParentReturn = parent::render();
+
+        if ($this->_isCustomPaymentMethod()) {
+            $sFnc = Registry::getRequest()->getRequestParameter('fnc');
+            $this->_aViewData["bConfigNotValid"] = ($sFnc === 'save' || $sFnc === 'addfield')
+                && !$this->_isSavePossible();
+        }
+
+        return $sParentReturn;
+    }
+
+    /**
+     * @inheritdoc
+     *
+     * @return void
+     */
+    public function save()
+    {
+        if ($this->_isCustomPaymentMethod() && !$this->_isSavePossible()) {
+            $sOxId = $this->getEditObjectId();
+            $this->setEditObjectId($sOxId);
+            return;
+        }
+
+        parent::save();
+    }
+
+    /**
+     * Checks if it is possible to save the config
+     *
+     * @return bool
+     */
+    private function _isSavePossible()
+    {
+        $aParams = Registry::getRequest()->getRequestParameter('editval');
+
+        $bCredentialsValid = $this->_isFormCredentialsValid($aParams);
+
+        $sCountryCode = $aParams['oxpayments__wdoxidee_countrycode'];
+        $bCountryCodeValid = $aParams['oxpayments__oxid'] !== SofortPaymentMethod::getName(true)
+            || (preg_match('/^[a-z]{2}_[a-z]{2}$/', $sCountryCode) === 1);
+
+        return $bCredentialsValid && $bCountryCodeValid;
+    }
+
+    /**
+     * Checks if the credentials from the payment method form are valid
+     *
+     * @param array $aParams
+     *
+     * @return bool
+     */
+    private function _isFormCredentialsValid($aParams)
+    {
+        $sUrl = $aParams['oxpayments__wdoxidee_apiurl'];
+        $sUser = $aParams['oxpayments__wdoxidee_httpuser'];
+        $sPass = $aParams['oxpayments__wdoxidee_httppass'];
+
+        if ($sUrl && $sUser && $sPass) {
+            $oConfig = new Config($sUrl, $sUser, $sPass);
+            $oTransactionService = new TransactionService($oConfig, Registry::getLogger());
+            return $oTransactionService->checkCredentials();
+        }
+
+        return false;
+    }
+
+    /**
+     * Checks if a custom payment method is shown
+     *
+     * @return bool
+     */
+    private function _isCustomPaymentMethod()
+    {
+        $sOxId = $this->getEditObjectId();
+        $oPayment = oxNew(Payment::class);
+        $oPayment->load($sOxId);
+
+        return isset($sOxId) && $sOxId != '-1' && $oPayment->isCustomPaymentMethod();
```

Comment:
`$oPayment->isCustomPaymentMethod()` is enough already, it will return false for "empty" payment methods.

Reply:
👍